src/Model/UserModel.ts

```diff
@@ -0,0 +1,144 @@
+import { router } from "yandex-maps";
+import { apiConfig } from "../modules/api";
+import { Api } from "../modules/api/src/api";
+import { EventDispatcher, Listenable } from "../modules/observer";
+
+export type User = {
+    Username: string;
+    Email: string;
+    Password: string;
+    Icon: string;
+};
+
+export const enum UserEvent {
+    USER_LOGIN = "USER_LOGIN",
+    USER_REG = "USER_REG",
+    USER_LOGOUT = "USER_LOGOUT",
+    USER_UPDATE = "USER_UPDATE",
+    AUTH = "AUTH",
+}
+
+/**
+ * Модель пользователя
+ * @class
+ */
+export class UserModel implements Listenable<UserEvent> {
+    /**
+     * Пользователь
+     */
+    private user: User | null;
+    private errorMsg: string | null;
+
+    private events_: EventDispatcher<UserEvent>;
+    get events(): EventDispatcher<UserEvent> {
+        return this.events_;
+    }
+
+    /**
+     * Конструктор
+     */
+    constructor() {
+        this.events_ = new EventDispatcher<UserEvent>();
+        this.user = null;
+    }
+
+    usersDiff(newUserData: { [index: string]: string }, oldUser: User | null) {
+        const userDiff = {};
+        if (oldUser) {
+            for (const [key, value] of Object.entries(oldUser)) {
+                if (newUserData[key] && newUserData[key] !== value) {
+                    userDiff[key] = newUserData[key];
+                }
+            }
+        }
```

Comment:
return с if'ом более читаемый, хотя и является антипаттерном структурного программирования
Но структурное программирование было придумано в иных реалиях и давно, поэтому можно и забить на него в этом моменте :)

```suggestion
        if (!oldUser) {
            return;
        }
            
       for (const [key, value] of Object.entries(oldUser)) {
                if (newUserData[key] && newUserData[key] !== value) {
                    userDiff[key] = newUserData[key];
                }
       }
```